clang/lib/Sema/SemaDecl.cpp

```diff
@@ -11465,6 +11465,10 @@ static bool CheckMultiVersionFirstFunction(Sema &S, FunctionDecl *FD) {
   // otherwise it is treated as a normal function.
   if (TA && !TA->isDefaultVersion())
     return false;
+  // The target_version attribute only causes Multiversioning if this
+  // declaration is NOT the default version.
+  if (TVA && TVA->isDefaultVersion())
+    return false;
```

Comment:
Is it allowed to have both `target` and `target_version` on the same function? If so, can these early returns interfere with each other?

Reply:
`error: 'target_version' and 'target' attributes are not compatible`

No it's not allowed, but that doesn't mean that the early exit conditions cannot interfere. For example as I explain below !TVA should not early exit because this function check the target attribute as well.